gutenberg/acquire/metadata.py

```diff
@@ -26,23 +28,148 @@
 from gutenberg._util.os import remove
 
 
-_METADATA_CACHE = local_path(os.path.join('metadata', 'metadata.db'))
-_METADATA_DATABASE_SINGLETON = None
-
-
-@contextlib.contextmanager
-def _download_metadata_archive():
-    """Makes a remote call to the Project Gutenberg servers and downloads the
-    entire Project Gutenberg meta-data catalog. The catalog describes the texts
-    on Project Gutenberg in RDF. The function returns a file-pointer to the
-    catalog.
-
-    """
-    data_url = r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
-    with tempfile.NamedTemporaryFile(delete=False) as metadata_archive:
-        shutil.copyfileobj(urllib2.urlopen(data_url), metadata_archive)
-    yield metadata_archive.name
-    remove(metadata_archive.name)
+_GUTENBERG_CATALOG_URL = \
+    r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
+
+
+class CacheAlreadyExistsException(Exception):
```

Comment:
@cpeel: What do you think about replacing these custom exceptions with `ValueError`? The exceptions can only be raised on one context so that using a built-in exception type is unambiguous.